src/styles/functions.scss

```diff
@@ -0,0 +1,23 @@
+
+$week-days: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+
+@each $day, $day-position in $week-days {
+  .calendar--start-day-#{$day} > :first-child {
+    margin-left: ($cell-size + $gap) * $day-position;
+  }
+}
+
+@for $i from 28 through 30 {
+  .calendar--month-length-#{$i} .calendar__day:nth-child(n + #{$i + 1}) {
```

Comment:
The selector `.calendar__day:nth-child(n + #{$i + 1})` assumes that `.calendar__day` elements are direct children of `.calendar--month-length-#{$i}`, which might not be the case if additional elements are added in the future. Consider adding a direct child combinator (>) to ensure the styles are applied only to the direct children.